Sources/protoc-gen-swift/EnumGenerator.swift

```diff
@@ -140,18 +140,29 @@ class EnumGenerator {
   private func generateProtoNameProviding(printer p: inout CodePrinter) {
     let visibility = generatorOptions.visibilitySourceSnippet
 
-    p.print("\(visibility)static let _protobuf_nameMap: \(namer.swiftProtobufModulePrefix)_NameMap = [")
+    p.print("""
+        #if swift(>=5.10)
+          \(visibility)static nonisolated(unsafe) let _protobuf_nameMap: \(namer.swiftProtobufModulePrefix)_NameMap = _makeNameMap()
```

Comment:
The fact that these might be public makes me a touch nervous. Why are these public again?

Reply:
I agree. Since they are public we cannot use `nonisolated(unsafe)` here since anybody could grab those; however, from looking at `_NameMap` this appears to be an immutable struct and the only reason that it isn't `Sendable` right now is due to the storage of a few `UnsafeRawBufferPointer`. We really should make the underlying type `Sendable` here by either coming up with an implementation that is `Sendable` by default or by double-checking that this is indeed immutable and then adding an `@unchecked Sendable` to the type.

Reply:
I think revisiting the impl given how `String` has changed since the Swift 3 days (which is where I believe @tbkka `_NameMap` work goes back to) is likely a good idea, that's why I opened #1561.

If we push ahead with that in the coming weeks, this would likely just be sorter lived.

As for why they are `public`, it's because they are an api point that the Runtime needs to be able to invoke, and I don't believe we ever found a better way to handle that – i.e. - generated protocol conformance that the core runtime can call things on.

Reply:
It's unfortunate that the need for these to satisfy a protocol witness forces them to be public. Ideally these conformances would be on a type that didn't need to be public, though I understand why they are.

Reply:
> Until then I would recommend using `@unchecked Sendable` on the type instead of `nonisolated(unsafe) let` on the global

Is that likely to have to ripple through the support types also?  i.e. - `Name`, `InternPool`?

I'm doing this somewhat blind (need to do Xcode updates to locally test), so this PR might have to sit a little while until so I can look at switching it over.